zerver/tests/test_users.py

```diff
@@ -460,6 +461,45 @@ def test_access_user_by_id(self) -> None:
             self.example_user("cordelia"), self.example_user("aaron").id, for_admin=False
         )
 
+    def test_access_user_by_id_including_cross_realm(self) -> None:
+        iago = self.example_user("iago")
+        internal_realm = get_realm(settings.SYSTEM_BOT_REALM)
+
+        # Must be a valid user ID in the realm
+        with self.assertRaises(JsonableError):
+            access_user_by_id_including_cross_realm(iago, 1234, for_admin=False)
+        with self.assertRaises(JsonableError):
+            access_user_by_id_including_cross_realm(
+                iago, self.mit_user("sipbtest").id, for_admin=False
+            )
+
+        # Can only access bot users if allow_bots is passed
+        bot = get_system_bot(settings.WELCOME_BOT, internal_realm.id)
+        access_user_by_id_including_cross_realm(iago, bot.id, allow_bots=True, for_admin=False)
+        with self.assertRaises(JsonableError):
+            access_user_by_id_including_cross_realm(iago, bot.id, for_admin=False)
```

Comment:
So this is verifying the main logic that's unique about this test compared to the one you copy-pasted. This should really really have have updated comments that make it clear what you're verifying, since it's NOT the same as what was present in the previous test.